presto-lark-sheets/src/main/java/com/facebook/presto/lark/sheets/api/SheetsApiFactory.java

```diff
@@ -0,0 +1,225 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.lark.sheets.api;
+
+import com.facebook.presto.lark.sheets.SheetsConfig;
+import com.facebook.presto.lark.sheets.SheetsErrorCode;
+import com.facebook.presto.lark.sheets.SheetsUtil;
+import com.facebook.presto.spi.PrestoException;
+import com.larksuite.oapi.core.AppSettings;
+import com.larksuite.oapi.core.AppType;
+import com.larksuite.oapi.core.Config;
+import com.larksuite.oapi.core.DefaultStore;
+import com.larksuite.oapi.core.Domain;
+import com.larksuite.oapi.core.api.response.Response;
+import com.larksuite.oapi.service.drive_permission.v2.DrivePermissionService;
+import com.larksuite.oapi.service.drive_permission.v2.model.PublicGetReqBody;
+import com.larksuite.oapi.service.drive_permission.v2.model.PublicGetResult;
+import com.larksuite.oapi.service.sheets.v2.SheetsService;
+import com.larksuite.oapi.service.sheets.v2.model.MetainfoProperties;
+import com.larksuite.oapi.service.sheets.v2.model.Sheet;
+import com.larksuite.oapi.service.sheets.v2.model.SpreadsheetsMetainfoResult;
+import com.larksuite.oapi.service.sheets.v2.model.SpreadsheetsValuesGetResult;
+
+import javax.annotation.concurrent.ThreadSafe;
+import javax.inject.Inject;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import static com.facebook.presto.lark.sheets.SheetsErrorCode.SHEETS_API_ERROR;
+import static com.facebook.presto.lark.sheets.SheetsUtil.loadAppSecret;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+@ThreadSafe
+public class SheetsApiFactory
+        implements Supplier<SheetsApi>
+{
+    private final DrivePermissionService drivePermissionService;
+    private final SheetsService sheetsService;
+
+    @Inject
+    public SheetsApiFactory(SheetsConfig config)
+    {
+        this(requireNonNull(config, "config is null").getAppDomain(), config.getAppId(), config.getAppSecretFile());
+    }
+
+    public SheetsApiFactory(SheetsConfig.Domain domain, String appId, String appSecretFile)
+    {
+        requireNonNull(appId, "appId is null");
+        requireNonNull(appSecretFile, "appSecretFile is null");
+
+        String appSecret = loadAppSecret(appSecretFile);
+        AppSettings appSettings = new AppSettings(AppType.Internal, appId, appSecret, null, null);
+        Config config = new Config(toLarkDomain(domain), appSettings, new DefaultStore());
+        drivePermissionService = new DrivePermissionService(config);
+        sheetsService = new SheetsService(config);
+    }
+
+    @Override
+    public SheetsApi get()
+    {
+        return new SimpleSheetsApi();
+    }
+
+    public class SimpleSheetsApi
+            implements SheetsApi
```

Comment:
Just wondering as the `CachingSheetsApi` is implemented in a separate file, why do you prefer to implement the `SimpleSheetsApi` in the factory class? Any specific reasons?

Reply:
Possible some of my perference during development I guess. Have rafactored SimpleSheetsApi to a top level class to make it matched the code style of CachingSheetsApi.